File: backend/game-app/src/test/kotlin/com/opensam/entity/SelectPoolTest.kt
package com.opensam.entity

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.time.OffsetDateTime

class SelectPoolTest {

    @Test
    fun `SelectPool entity initializes with defaults`() {
        val pool = SelectPool(worldId = 1L, uniqueName = "test-pool")
        assertEquals(1L, pool.worldId)
        assertEquals("test-pool", pool.uniqueName)
        assertNull(pool.ownerId)
        assertNull(pool.generalId)
        assertNull(pool.reservedUntil)
        assertTrue(pool.info.isEmpty())
    }

    @Test
    fun `SelectPool info stores general template data`() {
        val pool = SelectPool(
            worldId = 1L,
            uniqueName = "pool-1",
            info = mutableMapOf(
                "generalName" to "조조",
                "leadership" to 96,
                "strength" to 72,
                "intel" to 91,
                "politics" to 93,
                "charm" to 96,
            ),
        )
        assertEquals("조조", pool.info["generalName"])
        assertEquals(96, pool.info["leadership"])
    }

    @Test
    fun `SelectPool reservation tracks owner and expiry`() {
        val pool = SelectPool(worldId = 1L, uniqueName = "pool-1")
        val expiry = OffsetDateTime.now().plusMinutes(10)
        pool.ownerId = 42L
        pool.reservedUntil = expiry

        assertEquals(42L, pool.ownerId)
        assertNotNull(pool.reservedUntil)
    }
}

File: frontend/src/app/(admin)/admin/select-pool/select-pool.test.ts
import { describe, expect, it } from 'vitest';
import type { SelectPoolEntry } from '@/types';

describe('SelectPool admin page types', () => {
    it('SelectPoolEntry has required fields', () => {
        const entry: SelectPoolEntry = {
            id: 1,
            worldId: 1,
            uniqueName: 'pool-1',
            ownerId: null,
            generalId: null,
            reservedUntil: null,
            info: { generalName: '조조', leadership: 96, strength: 72, intel: 91, politics: 93, charm: 96 },
            createdAt: '2026-01-01T00:00:00Z',
        };
        expect(entry.uniqueName).toBe('pool-1');
        expect(entry.info.generalName).toBe('조조');
        expect(entry.info.leadership).toBe(96);
    });

    it('available pool entry has null generalId and ownerId', () => {
        const entry: SelectPoolEntry = {
            id: 2,
            worldId: 1,
            uniqueName: 'pool-2',
            ownerId: null,
            generalId: null,
            reservedUntil: null,
            info: {},
            createdAt: '2026-01-01T00:00:00Z',
        };
        expect(entry.generalId).toBeNull();
        expect(entry.ownerId).toBeNull();
    });

    it('picked pool entry has generalId set', () => {
        const entry: SelectPoolEntry = {
            id: 3,
            worldId: 1,
            uniqueName: 'pool-3',
            ownerId: 42,
            generalId: 100,
            reservedUntil: '2026-01-01T00:10:00Z',
            info: {},
            createdAt: '2026-01-01T00:00:00Z',
        };
        expect(entry.generalId).toBe(100);
    });
});
